data_layout.go

```diff
@@ -0,0 +1,421 @@
+package irutil
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// DataLayout is a structural representation of the datalayout
+// string from https://llvm.org/docs/LangRef.html#data-layout
+type DataLayout struct {
+	isBigEndian               bool                                   // default - True
+	naturalStackAlignment     uint64                                 // bits in multiple of 8, default: 0
+	programMemoryAddressSpace uint64                                 // default - 0
+	globalVarAddressSpace     uint64                                 // default - 0
+	allocaAddressSpace        uint64                                 // default - 0
+	ptrSizeAlign              map[uint64]*PointerSizeAlignment       // Key is addressspace
+	intSizeAlign              map[uint64]*IntegerSizeAlignment       // Key is size
+	vecSizeAlign              map[uint64]*VectorSizeAlignment        // Key is size
+	floatSizeAlign            map[uint64]*FloatingPointSizeAlignment // Key is size
+	aggAlign                  *AggregateAlignment                    // default - 0:64
+	funcPtrAlign              *FunctionPointerAlignment              // default - Not specified in LLVM Ref doc
+	mangling                  ManglingStyle                          // depends on os - "e" for linux, "o" for mac
+	nativeIntBitWidths        []uint64                               // default - depends on arch "8:16:32:64" for x86-64
+	nonIntegralPointerTypes   []string                               // Unsure of the datatype, so leaving it as string to accommodate anything.
+}
+
+// NewDataLayout returns an instance of DataLayout with default values
+// taken from https://llvm.org/docs/LangRef.html#data-layout
+func NewDataLayout(os string, arch string) *DataLayout {
```

Comment:
If possible, we should consider making `arch` an enum.

Reply:
@dannypsnl's comment looks better to me. If you guys can add the required enums there, I can just use it in this PR instead of opening another one.